collections/bitroomgrid.rs

```diff
@@ -0,0 +1,248 @@
+//! Dependencies: `serde`
+//!
+//! A 50×50 binary grid meant to be used to flag tiles in a room
+//!
+//! Has basic compression built in too for more compact serialization
+
+
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::ser::{self, Serialize, Serializer};
+use std::fmt::{self, Write};
+
+const ROWS: usize = 10;
+const ROW_SIZE: usize = 32 * 8; // Each row holds 32 * 8 = 256 bits of data
+
+/// 50×50 binary flag map
+#[derive(Debug, Clone)]
+pub struct BitRoomGrid {
+    buffer: [[u8; ROW_SIZE / 8]; ROWS],
+}
+
+impl Default for BitRoomGrid {
+    fn default() -> Self {
+        BitRoomGrid {
+            buffer: [[0; ROW_SIZE / 8]; ROWS],
+        }
+    }
+}
+
+impl BitRoomGrid {
+    pub fn get(&self, x: u16, y: u16) -> bool {
+        let row, col = Self::get_row_col(x,y);
+
+        let row = &self.buffer[row];
+        let cell = row[col / 8];
+
+        let col = col % 8;
+
+        cell & (1 << col) != 0
+    }
+
+    pub fn set(&mut self, x: u16, y: u16, value: bool) {
+        let row, col = Self::get_row_col(x,y);
+
+        let row = &mut self.buffer[row];
+        let cell = &mut row[col / 8];
+
+        let col = col % 8;
+
+        if value {
```

Comment:
The xor (`^`) would collapse both here:
https://doc.rust-lang.org/std/ops/trait.BitXor.html